app/components/dashboard/DashboardContainer.jsx

```diff
@@ -1,29 +1,44 @@
 import React from 'react';
+import { browserHistory } from 'react-router';
 import Dashboard from './Dashboard.jsx';
 import Sidebar from '../Sidebar.jsx';
 
 class DashboardContainer extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { filters: ['filter 1', 'filter 2', 'filter 3'],
+    this.state = {
+      filters: ['filter 1', 'filter 2', 'filter 3'],
       formulae: ['formula 1', 'formula 2', 'formula 3'],
-      formulaInfo: {
-        name: 'name',
-        createdin: 'createdin',
-        date: 'date',
-        authenticated: 'authenticated',
-      },
+      selectedFilter: null,
+      selectedFormula: null,
     };
     this.getFilters = this.getFilters.bind(this);
     this.getFormulae = this.getFormulae.bind(this);
+    this.onSelectFormula = this.onSelectFormula.bind(this);
+    this.onSelectFilter = this.onSelectFilter.bind(this);
+    this.onClickConfigure = this.onClickConfigure.bind(this);
```

Comment:
its all being used. when the formula is being selected it is sending the props to the sidebar to render the specific content of the formula. When a filter is clicked we need to send a request to server to recieve the formulas pretaining to that filter. We need to store what is being selected so so that we can give it a class for css.